src/cuda.rs

```diff
@@ -0,0 +1,98 @@
+pub type c_char = u8;
+pub type size_t = usize;
+
+cfg_if! {
+    if #[cfg(libc_core_cvoid)] {
+        pub use ::ffi::c_void;
+    } else {
+        // Use repr(u8) as LLVM expects `void*` to be the same as `i8*` to help
+        // enable more optimization opportunities around it recognizing things
+        // like malloc/free.
+        #[repr(u8)]
+        #[allow(missing_copy_implementations)]
+        #[allow(missing_debug_implementations)]
+        pub enum c_void {
+            // Two dummy variants so the #[repr] attribute can be used.
+            #[doc(hidden)]
+            __variant1,
+            #[doc(hidden)]
+            __variant2,
+        }
+    }
+}
+
+extern "C" {
+    /// Syscall to be used whenever the *assert expression produces a `false` value*.
+    ///
+    /// Syscall arguments:
+    /// * `message`: The pointer to the string that should be output.
+    /// * `file`: The pointer to the file name string associated with the assert.
+    /// * `line`: The line number associated with the assert.
+    /// * `function`: The pointer to the function name string associated with the assert.
+    /// * `char_size`: The size in bytes of the characters contained in the provided strings. The only supported character size is 1.
+    ///
+    /// Source:
+    /// [PTX Interoperability](https://docs.nvidia.com/cuda/ptx-writers-guide-to-interoperability/index.html#system-calls).
+    pub fn __assertfail(
+        message: *const c_char,
+        file: *const c_char,
+        line: i32,
+        function: *const c_char,
+        char_size: size_t,
```

Comment:
While the syscall API takes `char_size`, e.g., the API exposed by clang does not (this is always known at compile-time).

Reply:
Anyway, we need to emit the correct signature. It's up to higher level crates to provide a convenient API without specifying the `char_size`, I think.

Reply:
Sure. I guess my point was that maybe this API is too low-level even for `libc`. 

Me as a libc maintainer and myself as a Rust CUDA person are two different people :P